OrbitCore/CrashHandler.cpp

```diff
@@ -35,10 +35,39 @@ CrashHandler::CrashHandler(const std::string& dump_path,
 }
 
 //-----------------------------------------------------------------------------
-void CrashHandler::DumpWithoutCrash() {
+void CrashHandler::DumpWithoutCrash() const {
   crashpad::NativeCPUContext cpu_context;
   crashpad::CaptureContext(&cpu_context);
   crashpad_client_.DumpWithoutCrash(cpu_context);
 }
 
+#else
```

Comment:
Okay, let's try to do this without #ifdef:
```
namespace {
template<typename ResultingStringType = base::FilePath::StringType>
struct StringTypeConverter {
  ResultingStringType operator()(std::string_view src) const {
    return ResultingStringType{src};
  }
};

template<>
struct StringTypeConverter<std::wstring> {
  std::wstring operator()(const std::string& src) const {
    return s2ws(str);
  }
};
} // namespace
```

And then in `CrashHandler::CrashHandler()`:
```
...
const base::FilePath dump_file_path(StringTypeConverter{}(dump_path));
const base::FilePath handler_file_path(StringTypeConverter{}(handler_path));
...
```

This way you only need the CrashHandler constructor once and we also avoid the macros.

Reply:
The problem is that `base::FilePath::StringType` is not only difference: `asynchronous_start` is only available on Windows as well as `CrashpadClient::WaitForHandlerStart`. But I'll double check if we can use `asynchronous_start = false` on Windows.
One more difference in function `CrashpadClient::DumpWithoutCrash`: in Windows version it is
`static void DumpWithoutCrash(const CONTEXT& context);`
in Linux it is
`static void DumpWithoutCrash(NativeCPUContext* context);`

Reply:
Ahhh. I missed that. In this case I would prefer multiple smallish-ifdefs instead of a big one. Because the big one is repeating a lot of code and it could easily happen that one misses updating both ifdef-branches.

Reply:
It looks like in our case we can use `asynchronous_start = false` for Windows too (and `CrashpadClient::WaitForHandlerStart` isn't needed for `asynchronous_start = false`).
I added StringTypeConverter as you proposed, and leave `#ifdef` just in `DumpWithoutCrash`.